app/grandchallenge/serving/models.py

```diff
@@ -47,26 +53,78 @@ class Download(models.Model):
     )
 
 
-def get_component_interface_values_for_user(*, user):
-    job_inputs_query = get_objects_for_user(
-        user=user, perms="algorithms.view_job"
-    ).filter(inputs__pk__in=OuterRef("pk"))
+def get_component_interface_values_for_user(
+    *, user, civ_pk=None, interface=None, parent_object_type_choice=None
+):
+    extra_filter_kwargs = {}
+    if interface:
+        extra_filter_kwargs["interface"] = interface
+    if civ_pk:
+        extra_filter_kwargs["pk"] = civ_pk
+    if (
+        parent_object_type_choice is not None
+        and parent_object_type_choice not in ParentObjectTypeChoices
+    ):
+        raise ValueError(
+            f"Unknown parent object type choice: {parent_object_type_choice}"
+        )
 
-    job_outputs_query = get_objects_for_user(
-        user=user, perms="algorithms.view_job"
-    ).filter(outputs__pk__in=OuterRef("pk"))
+    civs = ComponentInterfaceValue.objects.filter(**extra_filter_kwargs)
 
-    display_set_query = get_objects_for_user(
-        user=user, perms="reader_studies.view_displayset"
-    ).filter(values__pk__in=OuterRef("pk"))
+    pks_for_filter = []
+    if parent_object_type_choice in (None, ParentObjectTypeChoices.JOB):
+        job_query = filter_by_permission(
```

Comment:
Why include `None` here and in the other 3 cases? Shouldn't we only get civs for jobs/ds/ai if that was specifically requested?

Reply:
Ah I see why, you still need to filter the civs by permission for the user.